### DISCUSSION TITLE
How to get existing workers for a queue for shutdown

### DISCUSSION THREAD

**Discussion by dwcarr** [taskforcesh/bullmq#2362] [General]
I know there is a `Queue.getWorkers`, but this returns a string of worker names.  How do I get the worker based on that name?  Trying to write my shutdown function and was hoping to be able to use `getWorkers`  to find all the workers to close.

**Comment by manast** [comment]
I am late in responding this but for future users: as you always should keep a reference to every worker instance (as in ```const worker = await new Worker(...)```), you should just call ```await worker.close()``` for gracefully shutdown. It does not make sense to call ```getWorkers``` as potentially the workers could be running in many different machines. More info here: https://docs.bullmq.io/guide/going-to-production#gracefully-shut-down-workers
